### PR TITLE
refactor(compiler-cli): remove unused class decorator downlevel code

### PR DESCRIPTION
**Pull request by devversion**
The decorator downlevel transform is never used for actual class decorators because Angular class decorators rely on immediate execution for JIT. Initially we also supported downleveling of class decorators for View Engine library output, but libraries are shipped using partial compilation output and are not using this transform anymore.

All usages of the transform explicitly set `skipClassDecorators = true`, so this PR is in practice removing
unused/dead code.

Note that the transform is exclusively used for JIT processing, commonly for test files to help ease temporal dead-zone/forward-ref issues. We can remove the class decorator downlevel logic to remove technical debt.

### CODE REVIEW

**Review by alan-agius4 — APPROVED** [review]

**Review by clydin — APPROVED** [review]

**Review by clydin — APPROVED** [review]
reviewed-for: tooling-cli-shared-api

**Review by alan-agius4 — APPROVED** [review]
reviewed-for: tooling-cli-shared-api

File: yarn.lock [hunk 1]
```diff
@@ -11539,6 +11532,13 @@ magic-string@^0.25.7:
   dependencies:
     sourcemap-codec "^1.4.8"
 
+magic-string@^0.27.0:
```
Comment (JoostK) [inline yarn.lock:11535]: Seems unrelated?

**Review by JoostK — APPROVED** [review]

File: yarn.lock (hunk 1 above)
Comment (devversion) [reply yarn.lock:11535]: yeah, but it happens on `main` when someone runs `yarn`, so I figured I'd include it.

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit c241f63e8d2717d1dbc8b6a71d04a7c1095a155c.
